ukvm/ukvm_module_net.c

```diff
@@ -178,8 +178,10 @@ static void hypercall_netwrite(struct ukvm_hv *hv, ukvm_gpa_t gpa)
     int ret;
 
     ret = write(netfd, UKVM_CHECKED_GPA_P(hv, wr->data, wr->len), wr->len);
-    assert(wr->len == ret);
-    wr->ret = 0;
+    if (wr->len == ret)
+        wr->ret = 0;
+    else
+        wr->ret = -1;
 }
```

Comment:
There are two problems here:

1. The _modus operandi_ for error handling in Solo5, the ukvm hypercall API and ukvm itself has not really been thought through. This needs solving, but is a hard problem to get right, hence the aggressive use of "fail fast" for now.
2. This specific problem which I would describe as "you get an obscure error message if the host side of the tap interface is down, despite the guest having started up correctly".

I think the right solution for (2), without going into (1), is that we check that the interface is indeed up in ukvm `tap_attach()` by adding a check for `IFF_UP` (and `IFF_RUNNING`?) at [ukvm_module_net.c:96](https://github.com/Solo5/solo5/blob/master/ukvm/ukvm_module_net.c#L96) and have the function return `ENETDOWN` if not.

I would leave the assertion on the write path alone for now.

What do you think?